frontend/src/App.tsx

```diff
@@ -53,32 +60,11 @@ const App = (): React.ReactElement => {
             <Switch>
               <Route exact path={Routes.LOGIN_PAGE} component={Login} />
               <Route exact path={Routes.SIGNUP_PAGE} component={Signup} />
+              <Route exact path={Routes.UNAUTHORIZED_PAGE} component={Unauthorized} />
               <PrivateRoute exact path={Routes.HOME_PAGE} component={Default} />
-              <PrivateRoute
-                exact
-                path={Routes.CREATE_ENTITY_PAGE}
-                component={CreatePage}
-              />
-              <PrivateRoute
-                exact
-                path={Routes.UPDATE_ENTITY_PAGE}
-                component={UpdatePage}
-              />
-              <PrivateRoute
-                exact
-                path={Routes.DISPLAY_ENTITY_PAGE}
-                component={DisplayPage}
-              />
-              <PrivateRoute
-                exact
-                path={Routes.EDIT_TEAM_PAGE}
-                component={EditTeamInfoPage}
-              />
-              <PrivateRoute
-                exact
-                path={Routes.HOOKS_PAGE}
-                component={HooksDemo}
-              />
+              <PrivateRoute exact path={Routes.MAGAZINE_REVIEW_PAGE} component={MagazineReview} />
+              <PrivateRoute exact path={Routes.PROFILE_PAGE} component={Profile} />
+              <AdminRoute exact path={Routes.ADMIN_DASHBOARD_PAGE} component={AdminDashboard} />
```

Comment:
So I noticed you made a seperate component called AdminRoute. I searched up role based routing and it seems you can just do it using Private Routes by adding a "requiredRoles" field. Here's an example: https://adarshaacharya.com.np/blog/role-based-auth-with-react-router-v6
Do you think you could do it this way?